packages/service-library/src/servicelib/rabbitmq/_client.py

```diff
@@ -142,12 +166,21 @@ async def _on_message(
                             if not await message_handler(message.body):
                                 await message.nack()
                     except Exception:  # pylint: disable=broad-exception-caught
-                        _logger.exception(
-                            "unhandled exception when consuming RabbitMQ message, "
-                            "this is catched but should not happen. "
-                            "Please check, message will be queued back!"
-                        )
-                        await message.nack()
+                        if (
+                            max_retries_upon_error is not None
+                            and retry_count < max_retries_upon_error
+                        ):
+                            message.headers[_HEADER_X_RETRY_COUNT] = retry_count + 1
+                            _logger.exception(
```

Comment:
I fyou retry, shouldn;t this be a warning? it should rather be an error when it goes over max_retries?

Reply:
Good point. I've added this to the new design.